docs/syntax.md

```diff
@@ -0,0 +1,667 @@
+```
+
+Which binds a variable to the return of an inline recursive function.
+The function `go` is available inside the bend body and calls it recursively.
+
+It is possible to initialize multiple variables:
+
+```python
+bend x = 1, y = 2 ... while condition(x, y, ...):
+```
+
+It is equivalent to this inline recursive function:
+
+```python
+def bend(x, y, ...):
+  if condition(x, y, ...):
+    ...
+    return ... bend(x, y, ...) ...
+  else:
+    return ...
+```
+
+### Do
+
+```python
+do Result.bind:
+  x <- safe_div(2, 0)
+  return x
+```
+
+A monadic do block.
+
+Where `x <- ...` performs a monadic operation.
+
+Other statements are allowed inside the `do` block.
+
+## Expressions
+
+### Variables
+
+```python
+some_var
+
+foo/bar
+```
+
+A variable can be anything matching the regex `[A-Za-z0-9_.-/]+`.
+
+A variable is a name for some immutable expression. It is possible to rebind variables with the same name.
+
+```python
+x = 1
+x = x + 1
+```
+
+### Lambdas
+
+```python
+x => x
+
+x => y => y
+```
+
+Lambdas represents anonymous inline functions, it can bind a variable and has an expression as body.
+
+### Unscoped Lambdas and Variables
+
+```python
+$x => $x
+```
+
+Like lambdas, with the exception that the variable starts with a `$` sign. Every unscoped variable in a function must have a unique name and must be used exactly once.
+
+Unscoped variables are not transformed and linearized like normal scoped variables.
+
+Read [using scopeless lambdas](/docs/using-scopeless-lambdas.md) to know more about.
+
+### Function Call
+
+```python
+callee(arg_1, arg_2, arg_n)
+```
+
+A call is written with a callee followed by a list of arguments.
+
+The effect of a function call is to substitute the callee with it's body and replace the arguments by the passed variables.
+
+Accepts partial applications.
+
+### Tuple
+
+```python
+(3, 9)
+```
+
+A Tuple is surrounded by `(` `)` and should contain 2 or more elements. Elements are separated by `,`.
+
+### Superposition
+
+```python
+{1 2 3}
+```
+
+A superposition of values is defined using `{` `}` with at least 2 expressions inside.
+
+### Numbers and Infix Operations
+
+Currently, bend supports 3 types of numbers: floats, integers and unsigned integers. All of then are 24 bit sized.
+
+```python
+f24 = +88.012
+
+i24 = -42
+
+u24 = 42
+```
+
+| Operation      | Syntax |
+|----------------|--------|
+| Addition       | x + y  |
+| Subtraction    | x - y  |
+| Multiplication | x * y  |
+| Division       | x / y  |
+| Equal          | x == y |
+| Not Equal      | x != y |
+| Less Than      | x < y  |
+| Greater Than   | x > y  |
+| Bitwise And    | x & y  |
+| Bitwise Or     | x | y  |
+| Bitwise Xor    | x ^ y  |
+
+### Character Literal
+
+```python
+'x'
+```
+
+A Character is surrounded with `'`. Accepts unicode characters, unicode escapes in the form '\u{hex value}' and is desugared to the unicode codepoint as an `u24`.
+
+Only supports unicode codepoints up to `0xFFFFFF`.
```

Comment:
Either add codepoints to the dictionary or change it into 'code points' (apparently both are correct)